app/Http/Controllers/Staff/DonationController.php

```diff
@@ -74,16 +74,31 @@ public function update(Request $request, int $id): \Illuminate\Http\RedirectResp
         $donation->status = ModerationStatus::APPROVED;
         $donation->starts_at = $now;
 
-        if ($donation->package->donor_value > 0) {
-            $donation->ends_at = $now->addDays($donation->package->donor_value);
-        } else {
+        if ($donation->package->donor_value === null) {
             $donation->ends_at = null;
+        } else {
+            $activeDonations = Donation::where('user_id', $donation->user_id)
+                ->where('status', ModerationStatus::APPROVED)
+                ->where('ends_at', '>', $now)
+                ->get();
```

Comment:
In the supposed case that a user has donated 10000+ times, we don't need to pull all those records into memory. Can you query just the record with the latest (or null) ends_at instead? Then comparing it to null can be used for if there's any active donation or not.